functional/FunctionalTest.java

```diff
@@ -247,6 +247,25 @@ public static void putObject_test6() throws Exception {
   }
 
   /**
+   * Test: putObject(String bucketName, String objectName, String contentType, InputStream body).
+   */
+  public static void putObject_test7() throws Exception {
+    System.out.println("Test: putObject(String bucketName, String objectName, InputStream body, "
+                       + "String contentType)");
+    String filename = createFile(3 * MB);
+    InputStream is = Files.newInputStream(Paths.get(filename));
+    client.putObject(bucketName, filename, is, customContentType);
+    is.close();
+    Files.delete(Paths.get(filename));
+    ObjectStat objectStat = client.statObject(bucketName, filename);
+    if (!customContentType.equals(objectStat.contentType())) {
+      throw new Exception("[FAILED] Test: putObject(String bucketName, String objectName, String contentType, "
+                          + "long size, InputStream body)");
+    }
+    client.removeObject(bucketName, filename);
+  }
+
```

Comment:
Can you also add a test case that will hit the multi-part upload flow, meaning filesize greater that 5 MiB

Reply:
@balamurugana Would it be possible to make it possible to reduce the multipart size for unknown sized streams? I want to avoid having to keep a lot data in memory, and will probably never have any files larger than 1 GB. Does the max object size or max number of multiparts have to be a constant value?

Reply:
@andersflemmen Part size 525MiB is chosen for unknown sized stream multipart by calculation.  As we know maximum object size is 5TiB and maximum part count is 10000.  With these numbers, 525MiB is chosen.

As multipart upload is automatic in minio-java SDK, user has no control on part size.  Supporting custom part size requires design discussion.

Is there a possibility to stage the object data and call `putObject()` with size?

Reply:
@balamurugana I see! Would be nice to have the option to set the maximum object size, as we do not have file sizes close to 5TiB. We want to limit memory usage, and stream the data all the way from user upload and into Minio, so we will probably have to implement the multipart upload ourselves if it can't be provided through the default API.